pool/error_pool.go

```diff
@@ -93,5 +102,9 @@ func (p *ErrorPool) addErr(err error) {
 			p.errs = errors.Append(p.errs, err)
 		}
 		p.mu.Unlock()
+
+		if p.errorCallBack != nil {
+			p.errorCallBack(err)
+		}
```

Comment:
I have also thought about that but if the callback is doing something time intensive it would block the whole pool from call the callback.
I don't really have a strong opinion about this, it's a matter communication to the user so they know how it behaves either way.

Reply:
Yeah, makes sense. I also don't feel very strongly. I think it's fine the way it is, but we should document that the callback must be safe to call from multiple goroutines.